src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -357,20 +357,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send_normal(const void *buf, MPI_Aint cou
     }
 
     fi_addr_t dest_addr = MPIDI_OFI_av_to_phys(addr, receiver_nic, vci_remote);
-    if (MPIR_CVAR_CH4_OFI_ENABLE_INJECT && data_sz <= MPIDI_OFI_global.max_buffered_send &&
-        !MPIDI_OFI_ENABLE_HMEM) {
-        if (MPIDI_OFI_ENABLE_DATA) {
-            MPIDI_OFI_CALL_RETRY(fi_tinjectdata(MPIDI_OFI_global.ctx[ctx_idx].tx,
-                                                send_buf, data_sz, cq_data, dest_addr, match_bits),
-                                 vci_local, tinjectdata);
-        } else {
-            MPIDI_OFI_CALL_RETRY(fi_tinject(MPIDI_OFI_global.ctx[ctx_idx].tx,
-                                            send_buf, data_sz, dest_addr, match_bits),
-                                 vci_local, tinject);
-        }
-        MPIR_T_PVAR_COUNTER_INC(MULTINIC, nic_sent_bytes_count[sender_nic], data_sz);
-        MPIDI_OFI_send_event(vci_src, NULL, sreq, MPIDI_OFI_REQUEST(sreq, event_id));
-    } else if (!is_huge_send) {
```

Comment:
I'm confused by the commit message for this change, since syncflag looks to be handled independently above. Still, I think that it makes sense if all inject usage is limited to the "lightweight" send path so would support this removal.

Reply:
Yes, I removed the path to use injection for sync messages. I think the sync requirement negates the latency benefit of injection. 

Were you saying the commit message "refactor" hides this "change"? If so, I can edit the commit message.

Reply:
No, I'm just not seeing how this block is related to sync send. `send_normal` handles noth sync and non-sync cases. From what I can tell, this block can execute without the syncflag being true. Maybe I am missing something.

Reply:
non-sync small message send is directed to `send_lightweight` already. Only small message sync send goes to `send_normal`.